### PR TITLE
Feature/saf 3477/text input

### PR DESCRIPTION
**Pull request by daiana-simplenight**
Text Input component

### CODE REVIEW

File: src/components/Inputs/TextInput/TextInput.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+
+import TextInput from '.';
+
+const title = 'SimplenightUI/';
```
Comment (NakaSimpleNight) [inline src/components/Inputs/TextInput/TextInput.stories.tsx:6]:
```suggestion
const title = 'SimplenightUI/Inputs';
```

File: src/components/Inputs/BaseInput/BaseInput.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+interface BaseInputProps {
+  type: 'text';
```
Comment (NakaSimpleNight) [inline src/components/Inputs/BaseInput/BaseInput.tsx:4]: I would add an id or name to use those properties. Think of this input for any possible form.

File: src/components/Inputs/TextInput/TextInput.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import BaseInput from '../BaseInput';
+
+interface TextInputProps {
+  size?: 'large' | 'small';
+  placeholder?: string;
+  state?: 'idle' | 'disabled' | 'error' | 'success';
+  value: string;
+  // eslint-disable-next-line no-unused-vars
```
Comment (NakaSimpleNight) [inline src/components/Inputs/TextInput/TextInput.tsx:9]: I wouldn't say I like these comments, I prefer to disable this from the config file if needed.

**Review by NakaSimpleNight — CHANGES_REQUESTED** [review]

File: src/components/Inputs/TextInput/TextInput.tsx (hunk 3 above)
Comment (daiana-simplenight) [reply src/components/Inputs/TextInput/TextInput.tsx:9]: There isn't a problem with this rule to turn it off in general, it's just not needed in this particular case.

**Review by enriquesimplenight — APPROVED** [review]

**Review by NakaSimpleNight — APPROVED** [review]

File: src/components/Inputs/BaseInput/BaseInput.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    success:
+      'text-dark-1000 border-green-1000 focus:ring-0 focus:border-green-1000',
+    disabled: 'text-dark-600 border-dark-300 bg-dark-200',
+  };
+
+  const isDisabled = state === 'disabled';
+
+  return (
+    <input
+      id={name}
```
Comment (NakaSimpleNight) [inline src/components/Inputs/BaseInput/BaseInput.tsx:40]: The idea was to use name to be the same as both properties name and id

**Review by NakaSimpleNight — CHANGES_REQUESTED** [review]

**Review by isabel-light-it — APPROVED** [review]

File: src/components/Inputs/BaseInput/BaseInput.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  type: 'text';
+}
+
+type BaseInputProps = BaseInputSpecificProps & GeneralProps;
+
+interface ColorsMap {
+  [key: string]: string;
+}
+
+const BaseInput = ({
```
Comment (isabel-light-it) [inline src/components/Inputs/BaseInput/BaseInput.tsx:14]: Should we consider the possibility of having icons both left and right on this component?

Comment (daiana-simplenight) [reply src/components/Inputs/BaseInput/BaseInput.tsx:14]: Yes the idea was to add those properties progressively as needed, like in the password input PR

**Review by isabel-light-it — APPROVED** [review]

**Review by NakaSimpleNight — APPROVED** [review]

**Pull request merged**